lib/skate/settings/test_group.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Skate.Settings.TestGroup do
+  alias Skate.Settings.Db.TestGroup, as: DbTestGroup
+  alias Skate.Settings.Db.User, as: DbUser
+
+  @type t() :: %__MODULE__{
+          id: integer(),
+          name: String.t(),
+          users: [DbUser.t()]
+        }
+
+  @enforce_keys [:id, :name, :users]
+
+  defstruct [:id, :name, users: []]
+
+  @spec create(String.t()) :: __MODULE__.t()
+  def create(name) do
+    %DbTestGroup{name: name}
+    |> Skate.Repo.insert!()
+    |> Skate.Repo.preload(:users)
+    |> convert_from_db_test_group()
+  end
+
+  @spec get_all() :: [t()]
+  def get_all() do
+    DbTestGroup
+    |> Skate.Repo.all()
+    |> Skate.Repo.preload(:users)
+    |> Enum.map(&convert_from_db_test_group(&1))
+  end
+
+  @spec update(t()) :: t()
+  def update(test_group) do
+    existing_test_group =
+      DbTestGroup |> Skate.Repo.get(test_group.id) |> Skate.Repo.preload(:test_group_users)
```

Comment:
question: could `existing_test_group` be nil?

Reply:
Theoretically it could be, but given that this is the `update` function I think it's safe to assume that we're being passed an existing test group that already exists in the database, and I'm generally okay with throwing an exception if that isn't true.

Reply:
That's fair! A non-blocking suggestion - in the unlikely case that the group didn't exist, using  `Skate.Repo.get!` here might return a clearer exception message by detecting the unexpected `nil` earlier in the function. Likely not needed though based on your point that it is safe to assume the group will exist, so it is unlikely we'll see an exception from this anyway!

Reply:
That's a good idea, actually - I'll make the switch.